portal-impl/src/com/liferay/portal/servlet/ComboServlet.java

```diff
@@ -299,6 +300,23 @@ protected void doService(
 			if (cacheEnabled && (modulePathsString != null) &&
 				!PropsValues.COMBO_CHECK_TIMESTAMP) {
 
+				if (PropsValues.COMBO_MAX_FILES != -1) {
```

Comment:
I'm not sure if this is going work, because it limits the size of the cache `_bytesArrayPortalCache` for all combo requests since the last restart, but the LPS is asking to limit the number of files per individual requests.

Reply:
This works for the first request but for the next, `_bytesArrayPortalCache` will contain the files from the previous request plus the files from the new one, so it's going to definitely hit the limit after some requests and prevent the cache from working forever.

Reply:
yep, probably we want to count the number non-empty entries that exist in `bytesArray`, no matter if we grab the `bytesArray` value from cache or compute it. That would give us the number of actually resolved files that combo plans to add to the response.

A simpler approach would be to use `modulePaths.length` as good approximation of the number of requested files. Now that we canonicalize the paths before adding them to `modulePathsSet` and then we build the `modulePaths` array from such set, `modulePaths.length` can be more than enough to limit this. 

As opposed to counting non-empty elements in `bytesArray`, `modulePaths.length`  includes all requested resources, no matter if they actually resolve to a real resource in the system or not.

Reply:
Yeah, good point. 

It's true that checking `modulePaths.length` wouldn't have worked before my other PR related to `ComboServlet` was merged. Maybe that was the problem why this looked a bit harder to fix that it should have been... :+1:

Reply:
@dsanz @izaera so instead of doing the `COMBO_MAX_FILES != -1` check, we should check if `modulePaths.length > 0`?